android-wave-recorder/src/main/java/com/github/squti/androidwaverecorder/WaveHeaderWriter.kt

```diff
@@ -24,15 +24,37 @@
 
 package com.github.squti.androidwaverecorder
 
+import android.content.Context
 import android.media.AudioFormat
+import android.net.Uri
 import java.io.File
+import java.io.FileInputStream
 import java.io.RandomAccessFile
 
-internal class WaveHeaderWriter(private val filePath: String, private val waveConfig: WaveConfig) {
+internal class WaveHeaderWriter(
+    private val filePath: String?,
+    private val waveConfig: WaveConfig
+) {
+    var fileUri: Uri? = null
+    var context: Context? = null
+
+    constructor(context: Context, fileUri: Uri?, waveConfig: WaveConfig) : this(null, waveConfig) {
+        this.fileUri = fileUri
+        this.context = context
+    }
 
     fun writeHeader() {
-        val inputStream = File(filePath).inputStream()
-        val totalAudioLen = inputStream.channel.size() - 44
+        val inputStream: FileInputStream
+        if (filePath != null) {
+            inputStream = File(filePath).inputStream()
+        } else {
+            val descriptor =
+                context!!.contentResolver.openFileDescriptor(fileUri!!, "rw")?.fileDescriptor
```

Comment:
assertion `!!` may be cause of the NullPointerException. It's better to add a null check.

Reply:
ok, I think I am changing API of this class a little bit with function overloading.